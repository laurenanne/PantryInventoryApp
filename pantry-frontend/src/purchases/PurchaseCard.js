import React from "react";
import { useHistory } from "react-router-dom";
import TableRow from "@mui/material/TableRow";
import TableCell from "@mui/material/TableCell";

function PurchaseCard({ purchaseId, date }) {
  const history = useHistory();

  if (date) {
    date = date.toString().slice(0, 10);
  }

  const handleRowClick = () => {
    history.push(`/purchases/${purchaseId}`);
  };

  return (
    <TableRow onClick={() => handleRowClick()} key={purchaseId}>
      <TableCell>{purchaseId}</TableCell>
      <TableCell>{date}</TableCell>
    </TableRow>
  );
}

export default PurchaseCard;
