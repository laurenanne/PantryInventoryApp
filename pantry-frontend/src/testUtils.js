import React from "react";
import UserContext from "./auth/UserContext";

const demoUser = {
  username: "tester",
  first_name: "testfirst",
  last_name: "testlast",
  email: "test@test.net",
  isAdmin: true,
};

const UserProvider = ({ children, currentUser = demoUser }) => (
  <UserContext.Provider value={{ currentUser }}>
    {children}
  </UserContext.Provider>
);

export { UserProvider };
