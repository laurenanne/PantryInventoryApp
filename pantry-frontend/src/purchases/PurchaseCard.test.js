import React from "react";
import { render } from "@testing-library/react";
import PurchaseCard from "./PurchaseCard";
import { UserProvider } from "../testUtils";

it("matches snapshot", function() {
  let purchase = { purchaseId: 2, date: "2023-10-08" };
  const { asFragment } = render(<PurchaseCard purchase={purchase} />);
  expect(asFragment()).toMatchSnapshot();
});
