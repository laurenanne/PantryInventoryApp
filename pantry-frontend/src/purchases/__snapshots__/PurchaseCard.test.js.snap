// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<DocumentFragment>
  <tr
    class="MuiTableRow-root css-1q1u3t4-MuiTableRow-root"
  >
    <td
      class="MuiTableCell-root MuiTableCell-sizeMedium css-bsph2u-MuiTableCell-root"
    />
    <td
      class="MuiTableCell-root MuiTableCell-sizeMedium css-bsph2u-MuiTableCell-root"
    />
  </tr>
</DocumentFragment>
`;
